From deed09cf7e5bba16ede2d535d3466101028b5cb6 Mon Sep 17 00:00:00 2001
From: Kaan Alsan <alsan@google.com>
Date: Thu, 22 Jan 2026 07:25:57 -0800
Subject: [PATCH] Check for valid TabCollectionNode before checking if a view is dragging

Fixed: 477344294
Change-Id: I5f973989085afdf548eabe097c184d8f80831bf8
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/7507272
Commit-Queue: Kaan Alsan <alsan@chromium.org>
Reviewed-by: Eshwar Stalin <estalin@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1572994}
---

--- a/chrome/browser/ui/views/tabs/vertical/vertical_tab_group_view.cc
+++ b/chrome/browser/ui/views/tabs/vertical/vertical_tab_group_view.cc
@@ -195,6 +195,9 @@ views::Widget* VerticalTabGroupView::Sho
 }
 
 bool VerticalTabGroupView::IsViewDragging(const views::View& child_view) const {
+  if (!collection_node_ || !collection_node_->GetController()) {
+    return false;
+  }
   return GetDragHandler().IsViewDragging(child_view);
 }
 
--- a/chrome/browser/ui/views/tabs/vertical/vertical_unpinned_tab_container_view.cc
+++ b/chrome/browser/ui/views/tabs/vertical/vertical_unpinned_tab_container_view.cc
@@ -127,6 +127,9 @@ views::ProposedLayout VerticalUnpinnedTa
 
 bool VerticalUnpinnedTabContainerView::IsViewDragging(
     const views::View& child_view) const {
+  if (!collection_node_ || !collection_node_->GetController()) {
+    return false;
+  }
   return GetDragHandler().IsViewDragging(child_view);
 }
 
